src/networking.c

```diff
@@ -2758,18 +2758,17 @@ NULL
     }
 }
 
-/* HELLO <protocol-version> [AUTH <user> <password>] [SETNAME <name>] */
+/* HELLO [protocol-version] [AUTH <user> <password>] [SETNAME <name>] */
 void helloCommand(client *c) {
-    long long ver;
+    long long ver = 0;
 
-    if (getLongLongFromObject(c->argv[1],&ver) != C_OK ||
-        ver < 2 || ver > 3)
-    {
+    if (c->argc >= 2 && getLongLongFromObject(c->argv[1],&ver) == C_OK &&
+            (ver < 2 || ver > 3)) {
         addReplyError(c,"-NOPROTO unsupported protocol version");
         return;
     }
 
-    for (int j = 2; j < c->argc; j++) {
+    for (int j = ver == 0 ? 1 : 2; j < c->argc; j++) {
```

Comment:
a nit: I would use !ver, since I think that conveys the message a bit clearer that there is no version provided. Same with bellow.

Reply:
I dont know if `!!ver` is a good idea... How do you feel?